content/en/platform/corda/4.12/enterprise/upgrade-guide.md

```diff
@@ -0,0 +1,273 @@
+
+For more information about TVU and its features, go to the [Transaction Validator Utility]({{< relref "node/operating/tvu/_index.md" >}}) section.
+
+The example use case in this guide validates the transactions without any additional options.
+
+1. Drain the node.
+   You must perform this step so there are no in-flight transactions when the node is stopped. Not draining the node could lead to these transactions not being checked by TVU, potentially causing issues later on.
+3. Stop the node.
+4. Place the TVU JAR into the root folder of the Corda 4.11 node.
+5. Run TVU:
+   ```
+   $ java -jar corda-tools-transaction-validator-4.12-RC01.jar
+   Starting
+   Total Transactions: 41
+   Waiting for all transactions to be processed...
+   Success, All transactions processed.
+   Total time taken: 3943ms
+   ```
+
+### Upgrade 4.11 CorDapps
+
+You must update all custom CorDapps being upgraded that are running on Corda 4.11 nodes so they use Java 17 and Kotlin 1.9. For steps on updating Cordapps, see [Upgrading a CorDapp to Corda Enterprise Edition 4.12]({{< relref "app-upgrade-notes-enterprise.md" >}}).
+
+#### Flow versioning
+You must annotate any flow that initiates other flows with the `@InitiatingFlow` annotation, which is defined as:
+
+```
+annotation class InitiatingFlow(val version: Int = 1)
+```
+
+The version property, which defaults to 1, specifies the flow’s version. You must increment this integer value whenever a flow release includes changes that are not backwards compatible. However, in this case, the flow version must stay the same to ensure backward compatibility with the flows running on previous Corda versions. However, even though in this case you are not incrementing the flow version, you must always increment the new 4.12 contract version, see [Contract versioning]({{< relref "#contract-versioning" >}}).
+
+#### Contract versioning
+
+When Corda 4.12 starts, it loads the CorDapp contract JAR. For version 4.12, the contract version must be higher than the contract version in the `legacy-contracts` JAR file. Therefore, you must increment the contract version by at least one.
+
+#### CorDapp minimum platform version
+
+The new CorDapp should inherit the same minimum platform version as Corda 4.12, which is 140. This is not 14, to account for minor updates to Corda 4.11.
+
+#### CorDapp signing
+
+Upon startup, a node verifies the signing key for the CorDapps it uses, preventing unwanted code from being executed. In Corda 4.12, nodes having both new and legacy versions of a contract CorDapp must have the same signing keys, otherwise the node will fail to start.
```

Comment:
```suggestion
Upon startup, a node verifies the signing key for the CorDapps it uses, preventing unwanted code from being executed. In Corda 4.12, nodes having both new and legacy versions of a contract CorDapp must have the same signing keys; otherwise the node will fail to start.
```